unstable test (?) panic on processing encoded plan

## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

see https://github.com/pingcap/tidb/pull/39248#issuecomment-1365904713

seems can't handle getFlatPlan returns nil?

### 2. What did you expect to see? (Required)

### 3. What did you see instead (Required)

### 4. What is your TiDB version? (Required)

<!-- Paste the output of SELECT tidb_version() -->

after we added a unit test, it is happened more frequently

```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]

	panic: runtime error: invalid memory address or nil pointer dereference

[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x3b3019c]



goroutine 86923 [running]:

github.com/pingcap/tidb/executor.(*ExecStmt).Exec.func1()

	executor/adapter.go:451 +0x417

panic({0x4a07140, 0x76f7ae0})

	GOROOT/src/runtime/panic.go:884 +0x212

github.com/pingcap/tidb/planner/core.NormalizeFlatPlan(0xc013502780?)

	planner/core/encode.go:265 +0x5c

github.com/pingcap/tidb/executor.getPlanDigest(0xc013502780)

	executor/adapter.go:1706 +0x7c

github.com/pingcap/tidb/executor.(*ExecStmt).observeStmtBeginForTopSQL(0xc019dd30e0, {0x5880110, 0xc01b189260})

	executor/adapter.go:1895 +0x9a

github.com/pingcap/tidb/executor.(*ExecStmt).Exec(0xc019dd30e0, {0x5880110, 0xc01b189260})

	executor/adapter.go:516 +0x8d4

github.com/pingcap/tidb/session.runStmt({0x58800a0, 0xc0000600b0}, 0xc01160f180, {0x5891ee0, 0xc019dd30e0?})

	session/session.go:2359 +0x6c9

github.com/pingcap/tidb/session.(*session).ExecuteStmt(0xc01160f180, {0x58800a0, 0xc0000600b0}, {0x5898030?, 0xc019dd2ff0?})

	session/session.go:2223 +0xca5

github.com/pingcap/tidb/testkit.(*TestKit).ExecWithContext(0xc013377810, {0x58800a0, 0xc0000600b0}, {0x4fa5a7a, 0x1d}, {0x0, 0x0, 0x0})

	testkit/testkit.go:321 +0x76e

github.com/pingcap/tidb/testkit.(*TestKit).MustQueryWithContext(0xc013377810, {0x58800a0, 0xc0000600b0}, {0x4fa5a7a, 0x1d}, {0x0, 0x0, 0x0})

	testkit/testkit.go:153 +0x145

github.com/pingcap/tidb/testkit.(*TestKit).MustQuery(0x1951739?, {0x4fa5a7a?, 0x47e850d00?}, {0x0?, 0xc005e09680?, 0x1972707?})

	testkit/testkit.go:147 +0x8d

github.com/pingcap/tidb/executor_test.TestAdmin.func3()

	executor/executor_test.go:5665 +0xd1

created by github.com/pingcap/tidb/executor_test.TestAdmin

	executor/executor_test.go:5661 +0x1c5b
```

--- a/executor/executor_test.go
+++ b/executor/executor_test.go
@@ -5556,6 +5556,8 @@ func TestAdmin(t *testing.T) {
 	}))
 	tk := testkit.NewTestKit(t, store)
 	tk.MustExec("use test")
+	tk2 := testkit.NewTestKit(t, store)
+	tk2.MustExec("use test")
 	tk.MustExec("drop table if exists admin_test")
 	tk.MustExec("create table admin_test (c1 int, c2 int, c3 int default 1, index (c1))")
 	tk.MustExec("insert admin_test (c1) values (1),(2),(NULL)")
@@ -5680,7 +5682,7 @@ func TestAdmin(t *testing.T) {
 		// check that the result set has no duplication
 		defer wg.Done()
 		for i := 0; i < 10; i++ {
-			result := tk.MustQuery(`admin show ddl job queries 20`)
+			result := tk2.MustQuery(`admin show ddl job queries 20`)
 			rows := result.Rows()
 			rowIDs := make(map[string]struct{})
 			for _, row := range rows {
@@ -5711,7 +5713,7 @@ func TestAdmin(t *testing.T) {
 		// check that the result set has no duplication
 		defer wg2.Done()
 		for i := 0; i < 10; i++ {
-			result := tk.MustQuery(`admin show ddl job queries limit 3 offset 2`)
+			result := tk2.MustQuery(`admin show ddl job queries limit 3 offset 2`)
 			rows := result.Rows()
 			rowIDs := make(map[string]struct{})
 			for _, row := range rows {

--- a/planner/core/encode.go
+++ b/planner/core/encode.go
@@ -262,6 +262,9 @@ type planDigester struct {
 
 // NormalizeFlatPlan normalizes a FlatPhysicalPlan and generates plan digest.
 func NormalizeFlatPlan(flat *FlatPhysicalPlan) (normalized string, digest *parser.Digest) {
+	if flat == nil {
+		return "", parser.NewDigest(nil)
+	}
 	selectPlan, selectPlanOffset := flat.Main.GetSelectPlan()
 	if len(selectPlan) == 0 || !selectPlan[0].IsPhysicalPlan {
 		return "", parser.NewDigest(nil)

